ttnn/cpp/ttnn/operations/data_movement/permute/device/kernels/compute/transpose_xw_rm_single_tile_size.cpp

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cstdint>
+
+#include "compute_kernel_api/eltwise_unary/eltwise_unary.h"
+#include "compute_kernel_api/transpose_wh.h"
+#include "compute_kernel_api/tilize.h"
+#include "compute_kernel_api/untilize.h"
+#include "compute_kernel_api/pack_untilize.h"
+
+namespace NAMESPACE {
+void MAIN {
+    constexpr uint32_t x_block_size = get_compile_time_arg_val(0);
+    constexpr uint32_t w_block_size = get_compile_time_arg_val(1);
+
+    uint32_t num_blocks = get_arg_val<uint32_t>(0);
+
+    constexpr auto cb_in = tt::CBIndex::c_0;
+    constexpr auto cb_tilize = tt::CBIndex::c_1;
+    constexpr auto cb_out = tt::CBIndex::c_2;
+
+    unary_op_init_common(cb_in, cb_out);
+
+    for (uint32_t n = 0; n < num_blocks; n++) {
+        // have to global init here, otherwise pcc is bad
+        // if n > 0, then some register isn't cleared and the output of tilize_block is garbage
+        unary_op_init_common(cb_in, cb_out);
+        // tilize input via unpack and then pack
+        tilize_init_short(cb_in, 1);
+
+        cb_wait_front(cb_in, x_block_size);
+        // results are correct according to unpacker here
+        cb_reserve_back(cb_tilize, 1);
+
+        // removing this line causes the output of tilize_block to be garbage in the second iteration
+        tilize_block(cb_in, 1, cb_tilize);  // tilize and pack into cb_tilize
+
+        // tile slice according to unpacker is garbage after tilize_block in the second iteration, missing an uninit?
```

Comment:
reading these comments - looks like we got many lines of workarounds. is it the case?

Reply:
And @rdjogoTT found the issue and all these comments are no longer needed

Reply:
What was the issue?

Reply:
pack_untilize_uninit is hard coded to a default cb_id = 16, which was cb_out0 before. Since the recommendation is now to use cb_ids in sequential order what worked with the old cb_id=16 does not work with the cb_id that I had. There's a clean up item that they're working on that removes the defaults since they're misleading.